### PR TITLE
feat: NFTAccess: logic and tests for getAccountScore

### PR DESCRIPTION
**Pull request by davidprepo**
https://www.notion.so/prepo/NFTAccessHook-SC-Mini-Spec-5bbf9f36b33f4005933272429b274740#bec5868c22604023a866a0936de80b3d

Parent: https://github.com/prepo-io/prepo-monorepo/pull/584

### CODE REVIEW

File: apps/smart-contracts/core/test/DepositHook.test.ts [hunk 1]
```diff
@@ -659,4 +659,73 @@ describe('=> DepositHook', () => {
...
+        await Promise.all(mintTxs)
+      }
+      expect(await collection.balanceOf(user.address)).to.eq(numHeld)
+
+      if (collectionScore > 0) {
+        await depositHook
+          .connect(deployer)
+          .setCollectionScores([collection.address], [collectionScore])
+      }
+      expect(await depositHook.getCollectionScore(collection.address)).to.eq(collectionScore)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/DepositHook.test.ts:683]: dont need to test this here

File: apps/smart-contracts/core/test/DepositHook.test.ts [hunk 2]
```diff
@@ -659,4 +659,73 @@ describe('=> DepositHook', () => {
...
+
+      expect(await depositHook.getAccountScore(user.address)).to.eq(2)
+    })
+
+    it('returns 1 if holding 1 NFT from collection with score = 0, 1 NFT from collection with score = 1', async () => {
+      await prepareNFTs(firstERC721, 1, 0)
+      await prepareNFTs(secondERC721, 1, 1)
+
+      expect(await depositHook.getAccountScore(user.address)).to.eq(1)
+    })
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/DepositHook.test.ts:729]: add tests for:
-holding no NFTs from collection with score > 0
-holding 0 NFTs from collection with score = 0 (no preparation step needed for this one)

File: apps/smart-contracts/core/test/DepositHook.test.ts [hunk 3]
```diff
@@ -659,4 +659,73 @@ describe('=> DepositHook', () => {
...
+
+    it('returns 2 if holding 1 NFT from 2 collections each with score = 1', async () => {
+      await prepareNFTs(firstERC721, 1, 1)
+      await prepareNFTs(secondERC721, 1, 1)
+
+      expect(await depositHook.getAccountScore(user.address)).to.eq(2)
+    })
+
+    it('returns 2 if holding multiple NFTs from 2 collections each with score = 1', async () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/DepositHook.test.ts:717]: use different scores for each collection (to differentiate) and just say `score > 0`

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

File: apps/smart-contracts/core/test/DepositHook.test.ts [hunk 4]
```diff
@@ -128,6 +128,43 @@ describe('=> DepositHook', () => {
       await depositHook.connect(deployer).setDepositRecord(mockDepositRecord.address)
     })
 
+    async function setupDepositPermissionForTest(
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/test/DepositHook.test.ts:131]:
```suggestion
    async function setupDepositPermissionForTest(
```

This was  left behind in the rebase, but is this necessary, given that it is renamed, modified in the next PR?

**Pull request merged**